src/go/rpk/pkg/config/schema.go

```diff
@@ -48,7 +48,12 @@ func (c *Config) FileLocation() string {
 	return c.fileLocation
 }
 
-type RedpandaConfig struct {
+// RedpandaNodeConfig is the source of truth for Redpanda node configuration.
+//
+// Cluster properties must NOT be enlisted in this struct. Adding a cluster
+// property here would cause the dependent libraries (e.g. operator) to wrongly
+// consider it a node property.
```

Comment:
NIT: Could we change this to discover cluster properties from running Redpanda instance using Admin API?

Reply:
No, because you need to set node properties in redpanda.yaml before starting the cluster. It may work now that redpanda.yaml accepts also cluster properties for backward compat, but this will change

Reply:
In other words we need to maintain Redpanda schema/types in go, right?

Maybe we could have some kind of automation to lint/vet latest schema?

Reply:
Not that I know, this is the ticket that I know but we haven't worked on it: https://github.com/redpanda-data/redpanda/issues/4899

Cc: @twmb

Reply:
I thought we specifically added a file that the operator could ingest to determine which properties are node properties. cc @jcsp